src/test/fixtures/profiles/processed-profile.js

```diff
@@ -161,11 +135,62 @@ export function addMarkersToThreadWithCorrespondingSamples(
       markersTable.phase.push(INTERVAL);
       markersTable.startTime.push(startTime);
       markersTable.endTime.push(maybeEndTime);
+      allTimes.add(maybeEndTime);
     }
+    allTimes.add(startTime);
     markersTable.data.push(payload);
     markersTable.category.push(0);
     markersTable.length++;
   });
+
+  // When the profile contains at least 1 sample, we use only the samples to
+  // control the initial range. Because of that we need to add samples so that
+  // the range includes these markers. Note that when a thread has no sample,
+  // then the markers are used to compute the initial range.
```

Comment:
Note:
We had that before. Then we implemented a behavior where we would use both markers AND samples to define the initial range. That's when we removed this fixture code. But _then_ we refined the behavior again, and the initial range is defined by samples __only__ if there is at least 1 sample, and otherwise by markers.

That's why I've added this again: for the new test, this is needed.

Reply:
Thanks for the info!